src/platform/tls_openssl.c

```diff
@@ -249,10 +253,47 @@ CxPlatTlsCertificateVerifyCallback(
         return FALSE;
     }
 
+    if (TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAGS_USE_PORTABLE_CERTIFICATES) {
+        if (Cert) {
+            PortableCertificate.Length = i2d_X509(Cert, &PortableCertificate.Buffer);
+            if (!PortableCertificate.Buffer) {
+                QuicTraceEvent(
```

Comment:
We should be bailing out in these error conditions.

Reply:
Updated.  The code will now fails if we fail to encode certificate. I left building the chain as soft failure e.g. we pass is much as we can to caller and let them to decide. .NET will try to download missing certificates if the chain is incomplete (or it may pick them up from cache)